Neos.Neos/Classes/Domain/Service/UserService.php

```diff
@@ -247,39 +247,21 @@ public function getCurrentUser(): ?User
             return null;
         }
 
-        $tokens = $this->securityContext->getAuthenticationTokens();
-        $user = array_reduce($tokens, function ($foundUser, TokenInterface $token) {
-            if ($foundUser !== null) {
-                return $foundUser;
-            }
-
-            /** @var ?Account $account */
+        foreach ($this->securityContext->getAuthenticationTokens() as $token) {
+            /** @var Account|null $account */
             $account = $token->getAccount();
             if ($account === null) {
-                return $foundUser;
+                continue;
             }
-
             $user = $this->getNeosUserForAccount($account);
-            if ($user === null) {
-                return $foundUser;
+            if ($user !== null) {
+                return $user;
             }
-
-            return $user;
-        }, null);
-
-        return $user;
-    }
-
-    public function getCurrentUserIdentifier(): ?UserId
-    {
-        $currentUser = $this->getCurrentUser();
-
-        return $currentUser
-            ? UserId::fromString($this->persistenceManager->getIdentifierByObject($currentUser))
-            : null;
+        }
+        return null;
     }
 
-    public function findByUserIdentifier(UserId $userId): ?User
```

Comment:
just checked, its not breaking to rename this one and remove `getCurrentUserIdentifier` as they are not part of 8.3